tempesta_fw/http_sess.c

```diff
@@ -207,6 +203,14 @@ tfw_http_sticky_build_redirect(TfwHttpReq *req, StickyVal *sv, RedirMarkVal *mv)
 	if (mv)
 		tfw_http_redir_mark_prepare(mv, m_buf, sizeof(m_buf), m_chunks,
 					    sizeof(m_chunks), &rmark);
+
+	sticky = req->vhost->cookie;
+	if (sticky->js_challenge) {
+		body = &sticky->js_challenge->body;
+		if (mv && tfw_strcpy(&sticky->js_challenge->rmark, &rmark))
```

Comment:
If many CPUs run the code, then they mangle the `rmark` by rewriting each other. We can not do copy here as I proposed in https://github.com/tempesta-tech/tempesta/pull/1448#pullrequestreview-483017256 because `tfw_h1_prep_redirect()` and `tfw_h2_prep_redirect()` -> `tfw_h2_append_predefined_body()` copy the body: we don't want extra copies. However, the functions accept `body` as `TfwStr`, so it seems we should do little bit more work here with `TfwCfgJsCh->body`: `body` should be allocated within `TfwCfgJsCh` as 3-chunks `TfwStr` (e.g. `rmark` becomes the middle chunk). This way we can just copy the 2-chunks TfwStr in each `tfw_http_sticky_build_redirect()` call (only `TfwStr` descriptors with different middle chunk pointers!) and pass the string descriptors for copying to the HTTP functions constructing the redirects.